pkg/runctl/controller.go

```diff
@@ -503,6 +545,12 @@ func (c *Controller) handlePipelineRunPrepare(
 	pipelineRun k8s.PipelineRun,
 	pipelineRunsConfig *cfg.PipelineRunsConfigStruct,
 ) (bool, error) {
+	l := utils.LoggerFromContext(ctx)
+	ctx = utils.NewLoggingContextWithValues(ctx, &l, "", getPipelineRunStatusForLogging(pipelineRun)...)
+
+	logger := utils.LoggerFromContext(ctx)
+	logger.V(3).Info("Preparing pipeline execution")
```

Comment:
Move into the if block below. Otherwise it would also be logged if no preparation takes place.